src/main/java/core/basesyntax/ReadWriteList.java

```diff
@@ -2,24 +2,38 @@
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.concurrent.locks.ReadWriteLock;
 
-public class ReadWriteList<E> {
-    private List<E> list = new ArrayList<>();
-    private ReadWriteLock lock = new ReentrantReadWriteLock();
+public class ReadWriteList<T> {
+    private final List<T> list = new ArrayList<>();
+    private final ReadWriteLock lock = new ReentrantReadWriteLock();
 
-    public void add(E element) {
-        // write your code here
+    public void add(T item) {
+        lock.writeLock().lock();
+        try {
+            list.add(item);
+            System.out.println("Added: " + item);
+        } finally {
+            lock.writeLock().unlock(); // Ensure the lock is released even if an exception occurs
+        }
     }
 
-    public E get(int index) {
-        // write your code here
-        return null;
+    public T get(int index) {
+        lock.readLock().lock();
+        try {
+            return list.get(index);
```

Comment:
Returning directly inside the try block can be risky because if the `return` statement throws an exception, the `finally` block may not execute as expected. Consider assigning the result to a local variable and returning after the finally block.